client-course-schedulizer/src/components/reuseables/Schedule/Schedule.tsx

```diff
@@ -79,14 +96,33 @@ const ScheduleBase = ({ calendarHeaders, groupedEvents, ...calendarOptions }: Sc
               {calenderHeadersNoEmptyInTerm.map((header) => {
                 return (
                   <div key={header} className="calendar-width hide-axis">
-                    <Calendar {...calendarOptions} key={header} events={filteredEvents[header]} />
+                    <Calendar
+                      {...calendarOptions}
+                      key={header}
+                      eventClick={handleEventClick}
+                      events={filteredEvents[header]}
+                    />
                   </div>
                 );
               })}
             </div>
           </Stick>
         </div>
       </div>
+      <Popover
```

Comment:
I think you'll want to make this a separate component and use it here and when you click the add button